### PR TITLE
[Merged by Bors] - Update dependencies `ron` `winit`& fix `cargo-deny` lists

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- there are a few new versions for `ron`, `winit`, `ndk`, `raw-window-handle`
- `cargo-deny` is failing due to new security issues / duplicated dependencies

## Solution

- Update our dependencies
- Note all new security issues, with which of Bevy direct dependency it comes from
- Update duplicate crate list, with which of Bevy direct dependency it comes from

`notify` is not updated here as it's in #2993

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
> error: failed to run custom build command for `wayland-sys v0.29.1`

> Caused by:
>   process didn't exit successfully: `/home/runner/work/bevy/bevy/target/debug/build/wayland-sys-465716ae7e3b3f27/build-script-build` (exit status: 101)
>   --- stdout
>   cargo:rerun-if-env-changed=WAYLAND_CLIENT_NO_PKG_CONFIG
>   cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
>   cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
>   cargo:rerun-if-env-changed=HOST_PKG_CONFIG
>   cargo:rerun-if-env-changed=PKG_CONFIG
>   cargo:rerun-if-env-changed=WAYLAND_CLIENT_STATIC
>   cargo:rerun-if-env-changed=WAYLAND_CLIENT_DYNAMIC
>   cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
>   cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
>   cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
>   cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
>   cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
>   cargo:rerun-if-env-changed=PKG_CONFIG_PATH
>   cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
>   cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
>   cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
>   cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
>   cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
>   cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
>   cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
>   cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
> 
>   --- stderr
>   thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: `"pkg-config" "--libs" "--cflags" "wayland-client"` did not exit successfully: exit status: 1

What's up with this CI error? Seems to be responsible for both failing steps?

File: deny.toml [hunk 1]
```diff
@@ -36,21 +40,26 @@ wildcards = "deny"
...
+    { name = "core-foundation", version = "0.7" },      # from winit v0.26.0
+    { name = "core-foundation-sys", version = "0.6" },  # from gilrs v0.8.1
+    { name = "core-foundation-sys", version = "0.7" },  # from winit v0.26.0
+    { name = "core-graphics", version = "0.19" },       # from winit v0.26.0
+    { name = "darling", version = "0.10" },             # from rodio v0.14.0
+    { name = "darling_core", version = "0.10" },        # from rodio v0.14.0
+    { name = "darling_macro", version = "0.10" },       # from rodio v0.14.0
+    { name = "mach", version = "0.2" },                 # from gilrs v0.8.1
+    { name = "mio", version = "0.7" },                  # from notify v5.0.0-pre.11
+    { name = "ndk", version = "0.3" },                  # from rodio v0.14.0
```
Comment (alice-i-cecile) [inline deny.toml:54]: Why does rodio itself pull in two versions of `ndk`?

Comment (mockersf) [reply deny.toml:54]: rodio is using oboe which depends on ndk 0.4 and cpal which depends on ndk 0.3. I only noted the dependency from Bevy, not the complete dependency path

**Comment by mockersf** [comment]
> What's up with this CI error? Seems to be responsible for both failing steps?

Building with all features on linux needs two extra dependencies: `libwayland-dev` and `libxkbcommon-dev`. I'm guessing the first is needed for wayland, not sure exactly about the second one. It would be nice if someone on linux could check and document on Bevy dependencies for linux

**Comment by mockersf** [comment]
ci failure is from a random network error, everything should be green now 👍

**Comment by parasyte** [comment]
Heads up! winit 0.26.0 breaks transparent windows on Windows 11, which will negatively affect #3105 

Upstream issue: https://github.com/rust-windowing/winit/issues/2082

**Comment by mockersf** [comment]
Would it fix transparency on linux (https://github.com/rust-windowing/winit/pull/2006)?

**Comment by parasyte** [comment]
It is mentioned in [the changelog](https://github.com/rust-windowing/winit/blob/master/CHANGELOG.md#0260-2021-12-01), so I think the update would fix transparent windows on X11.

**Comment by HeavyRain266** [comment]
> Building with all features on linux needs two extra dependencies: `libwayland-dev` and `libxkbcommon-dev`. I'm guessing the first is needed for wayland, not sure exactly about the second one. It would be nice if someone on linux could check and document on Bevy dependencies for linux

Package libxkbcommon is shared libinput wrapper used by X11 and Wayland as general input library which is definitely required to work with bevy.

**Comment by mockersf** [comment]
bors try

**Comment by cart** [comment]
Nice work!

**Pull request closed without merging**